music_assistant/providers/podcastfeed/__init__.py

```diff
@@ -0,0 +1,283 @@
+multiple instances with each one feed must exist.
+
+"""
+
+from __future__ import annotations
+
+from collections.abc import AsyncGenerator
+from io import BytesIO
+from typing import TYPE_CHECKING
+
+import aiohttp
+import podcastparser
+from music_assistant_models.config_entries import ConfigEntry, ConfigValueType
+from music_assistant_models.enums import (
+    ConfigEntryType,
+    ContentType,
+    ImageType,
+    MediaType,
+    ProviderFeature,
+    StreamType,
+)
+from music_assistant_models.errors import InvalidProviderURI, MediaNotFoundError
+from music_assistant_models.media_items import (
+    AudioFormat,
+    Episode,
+    ItemMapping,
+    MediaItemImage,
+    MediaItemType,
+    Podcast,
+    ProviderMapping,
+    SearchResults,
+)
+from music_assistant_models.streamdetails import StreamDetails
+
+from music_assistant.models.music_provider import MusicProvider
+
+if TYPE_CHECKING:
+    from music_assistant_models.config_entries import ProviderConfig
+    from music_assistant_models.provider import ProviderManifest
+
+    from music_assistant import MusicAssistant
+    from music_assistant.models import ProviderInstanceType
+
+CONF_FEED_URL = "feed_url"
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    # ruff: noqa: ARG001
+    if not config.get_value(CONF_FEED_URL):
+        msg = "No podcast feed set"
+        return InvalidProviderURI(msg)
+    return PodcastMusicprovider(mass, manifest, config)
+
+
+async def get_config_entries(
+    mass: MusicAssistant,
+    instance_id: str | None = None,
+    action: str | None = None,
+    values: dict[str, ConfigValueType] | None = None,
+) -> tuple[ConfigEntry, ...]:
+    """
+    Return Config entries to setup this provider.
+
+    instance_id: id of an existing provider instance (None if new instance setup).
+    action: [optional] action key called from config entries UI.
+    values: the (intermediate) raw values for config entries sent with the action.
+    """
+    return (
+        ConfigEntry(
+            key=CONF_FEED_URL,
+            type=ConfigEntryType.STRING,
+            default_value=[],
+            label="RSS Feed URL",
+            required=True,
+        ),
+    )
+
+
+class PodcastMusicprovider(MusicProvider):
+    """Podcast RSS Feed Music Provider."""
+
+    @property
+    def supported_features(self) -> set[ProviderFeature]:
+        """Return the features supported by this Provider."""
+        return {ProviderFeature.BROWSE, ProviderFeature.SEARCH, ProviderFeature.LIBRARY_PODCASTS}
+
+    async def handle_async_init(self) -> None:
+        """Handle async initialization of the provider."""
+        # ruff: noqa: S310
+        feed_url = podcastparser.normalize_feed_url(self.config.get_value(CONF_FEED_URL))
+        async with aiohttp.ClientSession() as session, session.get(feed_url) as response:
```

Comment:
```suggestion
        async self.mass.http_session.get(feed_url) as response:
```
Always re-use the existing global http session and not recreate new ones.